app/controllers/concerns/api_authenticated.rb

```diff
@@ -7,13 +7,20 @@ module ApiAuthenticated
   extend ActiveSupport::Concern
 
   included do
-    before_filter :authenticate_user!
+    before_filter :authenticate_user!, :get_authorized_resources
   end
 
   def authenticate_user!
     user_not_authorized unless current_user
   end
 
+  def get_authorized_resources
+    token_accounts = prx_auth_token.authorized_resources.try(:keys)
+    if token_accounts
+      current_user.approved_accounts ||= Account.where(id: token_accounts)
```

Comment:
Curious what @cavis might say about this as well - with the separation of accounts and users, where the token provides the list of accounts not the user, I think it makes sense to move these methods using the list of accounts out of the user model, perhaps into methods on the `ApiAuthenticated` concern, or just in the relevant controller?